wallet/src/lib.rs

```diff
@@ -443,10 +443,15 @@ mod wallet {
         struct In {
             settings: Option<CanisterSettings>,
         }
+        let mut settings = normalize_canister_settings(args.settings)?;
```

Comment:
Please take a look at create_wallet(), which calls this method settings specifically modified in order to have no controllers set.